src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -367,11 +368,12 @@ void ShenandoahControlThread::run_service() {
       last_shrink_time = current;
     }
 
-    // HEY! kemperw would like to have this thread sleep on a timed wait so it
-    // could be explicitly woken when there is something to do. The timed wait
-    // is necessary because this thread has a responsibility to send
-    // 'alloc_words' to the pacer when it does not perform a GC.
-    os::naked_short_sleep(ShenandoahControlIntervalMin);
+    {
+      // The timed wait is necessary because this thread has a responsibility to send
+      // 'alloc_words' to the pacer when it does not perform a GC.
+      MonitorLocker lock(&_control_lock, Mutex::SafepointCheckFlag::_no_safepoint_check_flag);
```

Comment:
This is just `Mutex::_no_safepoint_check_flag`, Hotspot uses that everywhere. Another instance in also in the same file.

Reply:
Sorry, my IDE does this. Fixed.